XMLtoMD/GuidelineXmlToMD/Program.cs

```diff
@@ -0,0 +1,179 @@
+                {
+
+                    IEnumerable<Guideline> guidelinesInSubsection = (from guideline in guidelinesInSections
+                                                                     where string.Equals(guideline.Subsection, subsection, StringComparison.OrdinalIgnoreCase)
+                                                                     select guideline);
+
+                    if (guidelinesInSubsection.Any())
+                    {
+                        Console.WriteLine($"     { subsection}");
+                        _MdWriter.WriteLine(subsection, format: MdFormat.Heading3, numNewLines: 1);
+
+                        foreach (Guideline guideline in guidelinesInSubsection)
+                        {
+                            _MdWriter.WriteUnorderedListItem(GetGuidelineEmoji(guideline) + " " + guideline.Text.Trim('"'), listIndent: 0);
+                        }
+                    }
+                    _MdWriter.WriteLine("", numNewLines: 1);
+
+                }
+            }
+        }
+
+        private static string GetGuidelineEmoji(Guideline guideline)
+        {
+            string emoji = "";
+            switch (guideline.Severity)
+            {
+                case "AVOID":
+                    emoji = ":no_entry:";
+                    break;
+                case "DO NOT":
+                    emoji = ":x:";
+                    break;
+                case "DO":
+                    emoji = ":heavy_check_mark:";
+                    break;
+                case "CONSIDER":
+                    emoji = ":grey_question:";
+                    break;
+
+                default:
+                    break;
+            }
+            return emoji;
+        }
+
+        private static void PrintSections(ICollection<Guideline> guidelines)
+        {
+            _MdWriter.WriteLine("Sections", format: MdFormat.Heading2);
+
+            List<string> subSections = new List<string>();
+
+            List<string> sections = GetSections(guidelines);
+            foreach (string section in sections)
+            {
+                Console.WriteLine(section);
+
+                _MdWriter.WriteUnorderedListItem(section, format: MdFormat.InternalLink, listIndent: 0);
+
+
+                subSections = (from guideline in guidelines
+                               where string.Equals(guideline.Section, section, System.StringComparison.OrdinalIgnoreCase)
+                               select guideline.Subsection).Distinct().OrderBy(x => x).ToList();
+
+                foreach (string subsection in subSections)
+                {
+                    Console.WriteLine($"     { subsection}");
+                    _MdWriter.WriteUnorderedListItem(subsection, format: MdFormat.InternalLink, listIndent: 1);
+                }
+                _MdWriter.WriteLine("", numNewLines: 1);
+
+
+
+            }
+
+
+
+        }
+
+        public static List<string> GetSections(ICollection<Guideline> guidelines)
```

Comment:
In general it is better to make methods like this be `IEnumerable` and if the callers want a list they can call `ToList` on the return value.